### ISSUE TITLE
if_switch_linter counts expressions incorrectly if the argument doesn't use '{'

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3042]
```r
l = if_switch_linter(max_branch_expressions = 2L)
lint('
  switch(x,
    a = foo(c, d, e),
    b = bar(f, g, h)
  )
', l)
# <text>:2:3: warning: [if_switch_linter] Prefer repeated if/else statements over overly-complicated switch() statements. Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
#   switch(x,
#   ^~~~~~~~~
```

The lint logic is written with `switch(a = { ... }, b = { ... })` in mind, whereas in this case the number of expressions is just 1.
